### PR TITLE
Adds spring-boot-starter to all Spring Cloud GCP Boot starters

### PR DESCRIPTION
**Pull request by joaoandremartins**
As recommended in https://docs.spring.io/spring-boot/docs/current-SNAPSHOT/reference/htmlsingle/#boot-features-custom-starter-module-starter

### CODE REVIEW

**Comment by joaoandremartins** [comment]
@snicoll FYI

**Comment by snicoll** [comment]
@joaoandremartins that's not how I would do this. I prefer things to be explicit and not redudant if possible. 

Besides, if we follow our chain of thoughts of having a `spring-cloud-gcp-starter`(that itself brings `spring-cloud-starter`), this PR will be irrelevant anyway: all GCP starters (besides the main one of course) would bring `spring-cloud-gcp-starter` and that one itself will bring the base stuff. This is the same pattern that we use [in Spring  Boot](https://github.com/spring-projects/spring-boot/blob/bee5fa7fc625e3878523834240c4e85746b2cd0c/spring-boot-project/spring-boot-starters/spring-boot-starter-web/pom.xml#L19)

File: spring-cloud-gcp-starters/spring-cloud-gcp-starter-core/pom.xml [hunk 1]
```diff
@@ -44,5 +44,10 @@
 			<groupId>org.springframework.boot</groupId>
 			<artifactId>spring-boot-autoconfigure</artifactId>
```
Comment (snicoll) [inline spring-cloud-gcp-starters/spring-cloud-gcp-starter-core/pom.xml]: this dependency is no longer necessary now.

**Comment by joaoandremartins** [comment]
Currently, all our other starters already import `spring-cloud-gcp-starter-core`, so that should be sorted out.
`spring-cloud-gcp-starter-core` should be regarded as our core starter. What is required for it to be in a proper state?

**Comment by snicoll** [comment]
> Currently, all our other starters already import spring-cloud-gcp-starter-core

That's not the case in the branch you are working on for the autoconfigure stuff. Perhaps rebasing would be in order?

**Comment by joaoandremartins** [comment]
My goal is to apply this in master first (to become effective immediately) and then rebase/merge with the `autoconfigure` branch.

**Comment by snicoll** [comment]
@joaoandremartins OK!

**Pull request merged**